addons/sourcemod/scripting/shavit-replay.sp

```diff
@@ -1761,52 +1808,90 @@ public void Shavit_OnLeaveZone(int client, int type, int track, int id, int enti
 	}
 }
 
-public void Shavit_OnFinish(int client, int style, float time, int jumps, int strafes, float sync, int track)
+public void Shavit_OnFinish(int client, int style, float time, int jumps, int strafes, float sync, int track, float oldtime, float perfs, float avgvel, float maxvel, int timestamp)
 {
-	if(Shavit_IsPracticeMode(client))
-	{
-		return;
-	}
-
-	if(!gCV_Enabled.BoolValue || (gCV_TimeLimit.FloatValue > 0.0 && time > gCV_TimeLimit.FloatValue))
+	if(Shavit_IsPracticeMode(client) || !gCV_Enabled.BoolValue || gI_PlayerFrames[client] == 0)
 	{
 		return;
 	}
 
+	bool isTooLong = (gCV_TimeLimit.FloatValue > 0.0 && time > gCV_TimeLimit.FloatValue);
+	bool isWR = true;
+	
 	float length = GetReplayLength(style, track);
 
 	if(gA_FrameCache[style][track].bNewFormat)
 	{
 		if(length > 0.0 && time > length)
 		{
-			return;
+			isWR = false;
 		}
 	}
-
 	else
 	{
 		float wrtime = Shavit_GetWorldRecord(style, track);
 
 		if(wrtime != 0.0 && time > wrtime)
 		{
-			return;
+			isWR = false;
 		}
 	}
 
-	if(gI_PlayerFrames[client] == 0)
+	Action action = Plugin_Continue;
+	Call_StartForward(gH_ShouldSaveReplayCopy);
+	Call_PushCell(client);
+	Call_PushCell(style);
+	Call_PushCell(time);
+	Call_PushCell(jumps);
+	Call_PushCell(strafes);
+	Call_PushCell(sync);
+	Call_PushCell(track);
+	Call_PushCell(oldtime);
+	Call_PushCell(perfs);
+	Call_PushCell(avgvel);
+	Call_PushCell(maxvel);
+	Call_PushCell(timestamp);
+	Call_PushCell(isTooLong);
+	Call_PushCell(isWR);
+	Call_Finish(action);
```

Comment:
you grab the return value of the forward but don't use it.

Reply:
it's used `bool makeCopy = (action != Plugin_Continue);`

Reply:
be sure to follow the general pattern that the rest of the timer uses. While assigning it to a variable is more verbose it's better to be consistent.
```
	Action result = Plugin_Continue;
	Call_StartForward(gH_Forwards_FinishPre);
	Call_PushCell(client);
	Call_PushArrayEx(snapshot, sizeof(timer_snapshot_t), SM_PARAM_COPYBACK);
	Call_Finish(result);

	if(result != Plugin_Continue && result != Plugin_Changed)
	{
		return;
	}
	```

Reply:
Also since it's a forward/event a more fitting naming would be Shavit_OnReplayCopy or Shavit_OnBackupReplay

Reply:
> be sure to follow the general pattern that the rest of the timer uses. While assigning it to a variable is more verbose it's better to be consistent.

It doesn't make it verbose, it makes it obvious what is happening.

> Also since it's a forward/event a more fitting naming would be Shavit_OnReplayCopy or Shavit_OnBackupReplay

Which forward name is this about? Shavit_OnReplaySaved runs for whenever a replay is saved, whether it's a copy or not. And I think Shavit_ShouldSaveReplayCopy is a fine name since it's about whether a copy of a replay should be saved. Those two names you offered don't match the forwards as well.